### ISSUE TITLE
Minimap expects minimap similar to xcode 11

### ISSUE THREAD

**Issue by crper** [microsoft/vscode#74891]
### The function definition is all suspended, which can be located quickly.

![](https://i.loli.net/2019/06/05/5cf7a892590b853392.png)

**Comment by sean-mcmanus** [comment]
This recently came up again for C/C++ in [microsoft/vscode-cpptools#14717](https://github.com/microsoft/vscode-cpptools/issues/14717).

I investigated whether this could be implemented in the C/C++ extension. The extension already provides hierarchical document symbols, including functions and methods, but minimap section-header detection and rendering are implemented in VS Code core. The extension API does not expose a way for an extension to add minimap section-header text or control its font, alignment, or positioning.

I am on the Microsoft C/C++ extension team and may be able to contribute a focused implementation to VS Code. Would the editor/minimap owners currently consider a PR that:

* obtains document symbols through VS Code’s existing language-feature infrastructure;
* displays function and method names as optional minimap labels;
* keeps the feature disabled by default initially; and
* includes focused tests and avoids requesting symbols when the option is disabled?

Before starting, I would appreciate guidance on whether this remains aligned with the intended minimap UX and whether document symbols are the preferred source.

**Comment by cD1rtX3** [comment]
I think my idea of having functions just be highlighed instead of having a button to jump keeps this as unintrusive as possible.

**Comment by Bishoy334** [comment]
It'd be nice to have some configuration to specify certain regexes and how the matches would look in the minimap and more control over the style of what is rendered in the minimap (i.e font size, opacity, weight, colour potentially??). 

I'm happy with it solely being the style as @cD1rtX3 mentioned to make longer files more readable
